src/overlays/actors/ovl_En_Viewer/z_en_viewer.c

```diff
@@ -34,24 +35,124 @@ const ActorInit En_Viewer_InitVars = {
     (ActorFunc)EnViewer_Draw,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/EnViewer_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/EnViewer_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/EnViewer_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/func_8089F014.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/func_8089F0A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/func_8089F17C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/func_8089F218.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/func_8089F2C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/EnViewer_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Viewer/EnViewer_Draw.s")
+u32 D_8089F4D0;
+
+void EnViewer_SetupAction(EnViewer* this, EnViewerActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnViewer_Init(Actor* thisx, PlayState* play) {
+    EnViewer* this = THIS;
+
+    this->unk_154 = D_8089F3E0;
+    D_8089F3E0++;
+    switch (ENVIEWER_GET_3(&this->actor)) {
+        case 0:
+            this->unk_148 = ENVIEWER_GET_FFFF(&this->actor) * 40.0f;
```

Comment:
i can't get any of these to work, please write examples

Reply:
`ENVIEWER_GET_FFFF` is weird since the cast happens before the shift, while the rest shift after which I'm not sure how to reconcile. However agreed the casts should just be apart of the macros for example `ENVIEWER_GET_7F` would be defined as 
```
#define ENVIEWER_GET_7F(thisx) ((u8)(((thisx)->params >> 2) & 0x7F))
```

Reply:
This is very strange, the first one matches with `this->unk148 = ((((u16)this->actor.params >> 9))) * 40.0f;`, but none of the others.